rasa/nlu/featurizers/count_vectors_featurizer.py

```diff
@@ -160,26 +175,31 @@ def __init__(
         # declare class instance for CountVectorizer
         self.vectorizer = vectorizer
 
-    def _get_message_text(self, message):
-        if message.get("spacy_doc"):  # if lemmatize is possible
-            text_tokens = [t.lemma_ for t in message.get("spacy_doc")]
-            # text = " ".join([t.lemma_ for t in message.get("spacy_doc")])
-        elif message.get("tokens"):  # if directly tokens is provided
-            text_tokens = [t.text for t in message.get("tokens")]
+    def _get_message_text_by_attribute(self, message, attribute="text"):
+
+        attribute = "" if attribute == "text" else attribute + "_"
+        if message.get("{0}spacy_doc".format(attribute)):  # if lemmatize is possible
+            tokens = [t.lemma_ for t in message.get("{0}spacy_doc".format(attribute))]
+        elif message.get(
+            "{0}tokens".format(attribute)
+        ):  # if directly tokens is provided
+            tokens = [t.text for t in message.get("{0}tokens".format(attribute))]
         else:
-            text_tokens = message.text.split()
+            return ""
 
-        text = re.sub(r"\b[0-9]+\b", "__NUMBER__", " ".join(text_tokens))
+        text = re.sub(r"\b[0-9]+\b", "__NUMBER__", " ".join(tokens))
         if self.lowercase:
             text = text.lower()
 
         if self.OOV_token and self.analyzer == "word":
             text_tokens = text.split()
-            if hasattr(self.vectorizer, "vocabulary_"):
+            if hasattr(self.vectorizer[attribute], "vocabulary_"):
```

Comment:
`attribute` is either `""` or, for example, `intent_`. Does not seems to be a good key. Can we change that?